### PR TITLE
25.1 create new task sidletskyi

### PR DESCRIPTION
**Pull request by SidletskyiOlexandr**
Happy Flow (onSwagger):

1. Log in
2. Make sure that you are owner or manager or member of him
3. Send the HTTP POST request to "task/addtask" without user assign
4. Send the HTTP POST request to "task/addtask" with multiple user assign

### CODE REVIEW

File: Provis.Core/Validation/CreateTaskValidation .cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+            RuleFor(task => task.DateOfEnd)
+                .NotEmpty()
+                .NotNull()
+                .GreaterThan(DateTime.UtcNow);
+
+            RuleFor(task => task.WorkspaceId)
+                .NotEmpty()
+                .NotNull();
+        }
```
Comment (AndriyKovalets) [inline Provis.Core/Validation/CreateTaskValidation .cs:34]: Add rule for status id for checking if it exist in TaskStatuses enum

File: Provis.Core/Services/TaskService.cs [hunk 2]
```diff
@@ -55,5 +65,50 @@ public async Task<List<TaskDTO>> GetUserTasksAsync(string userId, int workspaceI
...
+                    if (userTasks.Exists(x => x.UserId == item.UserId))
+                    {
+                        throw new HttpException(System.Net.HttpStatusCode.Forbidden,
+                            "This user has already assigned");
+                    }
+                    userTasks.Add(new UserTask
+                    {
+                        TaskId = task.Id,
+                        UserId = item.UserId,
+                        UserRoleTagId = item.RoleTagId
```
Comment (AndriyKovalets) [inline Provis.Core/Services/TaskService.cs:105]: Create mapper dor in

File: Provis.WebApi/Controllers/TaskController.cs [hunk 3]
```diff
@@ -25,5 +27,15 @@ public async Task<IActionResult> GetUserTasks(string userId, int workspaceId)
 
             return Ok(getTasks);
         }
+
+        [Authorize]
+        [HttpPost]
+        [Route("addtask")]
+        public async Task<IActionResult> AddTaskAsync([FromBody] TaskCreateDTO createDTO)
```
Comment (AndriyKovalets) [inline Provis.WebApi/Controllers/TaskController.cs:37]: why don't you use policy?

File: Provis.Core/Services/TaskService.cs [hunk 4]
```diff
@@ -55,5 +65,50 @@ public async Task<List<TaskDTO>> GetUserTasksAsync(string userId, int workspaceI
             var mapTask = _mapper.Map<List<TaskDTO>>(userTasks);
             return mapTask;
         }
+
+        public async Task CreateTaskAsync(TaskCreateDTO taskCreateDTO, string userId)
```
Comment (YevheniiVovk) [inline Provis.Core/Services/TaskService.cs:69]: Create tech dept ticket for make transaction logic and integrate it here

Comment (SidletskyiOlexandr) [reply Provis.Core/Services/TaskService.cs:69]: https://app.clickup.com/t/1tf8jxe

**Review by YevheniiVovk — APPROVED** [review]

**Review by AndriyKovalets — APPROVED** [review]

**Pull request merged**